servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/Publisher.java

```diff
@@ -1801,6 +1802,33 @@ public final Publisher<T> whenCancel(Runnable onCancel) {
         return beforeCancel(onCancel);
     }
 
+    /**
+     * Validate the outstanding demand ({@link Subscription#request(long)} minus
+     * {@link Subscriber#onNext(Object)}) does not go negative.
+     * @return The new {@link Publisher}.
```

Comment:
Consider clarifying what will happen when invalid demand is detected. Would be nice to also have some pseudo code that will demonstrate the flow, like we do for other operators.

Reply:
Will be also great to clarify a use-case for this operator.